.github/workflows/build-and-test-on-pr-events.yml

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License
+
+name: Pull Request Build and Test Workflow
+
+# Run this workflow every time a new pull request is created in the repository
+on:
+  push:
+    branches:
+      # Push events on the 2.4-li branch
+      - '2.4-li'
+      # Push events on the 2.4-li-dev-* branch
+      - '2.4-li-dev-*'
+  pull_request:
+    types: [opened, reopened, edited, synchronize]
```

Comment:
Does "edited" mean editing the description of the PR without changing the code?
If so, I feel we don't need to trigger the workflow upon "edited" events.

Reply:
I agree that we do not need to trigger this workflow upon `edit` events.